test/python/transpiler/test_light_cone.py

```diff
@@ -0,0 +1,294 @@
+        For that, it suffices to move the binding before comparison instead of having it
+        before the `PassManager` run.
+        """
+
+        num_qubits = 4
+        qc = real_amplitudes(num_qubits, entanglement="pairwise", reps=3).decompose()
+        pm = PassManager([LightCone(Pauli("Y" + ("I" * (num_qubits - 2)) + "Y"))])
+        params = np.random.rand(16)
+        qc.assign_parameters(params, inplace=True)
+        new_circuit = pm.run(qc)
+
+        qr = QuantumRegister(num_qubits)
+        expected = QuantumCircuit(qr)
+
+        expected.ry(params[0], qr[0])
+        expected.ry(params[1], qr[1])
+        expected.ry(params[2], qr[2])
+        expected.ry(params[3], qr[3])
+        expected.cx(qr[0], qr[1])
+        expected.cx(qr[2], qr[3])
+        expected.ry(params[4], qr[0])
+        expected.cx(qr[1], qr[2])
+        expected.ry(params[5], qr[1])
+        expected.ry(params[6], qr[2])
+        expected.ry(params[7], qr[3])
+        expected.cx(qr[0], qr[1])
+        expected.cx(qr[2], qr[3])
+        expected.ry(params[8], qr[0])
+        expected.cx(qr[1], qr[2])
+        expected.ry(params[9], qr[1])
+        expected.ry(params[10], qr[2])
+        expected.ry(params[11], qr[3])
+        expected.cx(qr[0], qr[1])
+        expected.cx(qr[2], qr[3])
+
+        self.assertEqual(Operator(expected), Operator(new_circuit))
+
+    def test_all_commuting(self):
+        """Test the LightCone pass for a circuit that fully commutes with an observable."""
+
+        num_qubits = 4
+        qr = QuantumRegister(num_qubits)
+        qc = QuantumCircuit(qr)
+        pm = PassManager([LightCone(Pauli("I" * (num_qubits - 1) + "Z"))])
+
+        qc.s(qr[0])
+        qc.z(qr[0])
+        qc.h(qr[2])
+        qc.cx(qr[0], qr[1])
+        qc.cx(qr[2], qr[3])
+
+        new_circuit = pm.run(qc)
+
+        self.assertEqual(sum(new_circuit.count_ops().values()), 0)
+
+    def test_commuting_block(self):
+        """Test the LightCone pass for a commuting block: currently gates are checked
+        one by one and commuting blocks are thus ignored.
+        """
+
+        num_qubits = 4
+        qr = QuantumRegister(num_qubits + 1)
+        qc = QuantumCircuit(qr)
+        pm = PassManager([LightCone(Pauli("I" * (num_qubits - 2) + "X" + "I" * (num_qubits - 2)))])
+
+        qc.cx(qr[2], qr[1])
+        qc.cx(qr[3], qr[4])
+        qc.cx(qr[1], qr[0])
+        qc.cx(qr[2], qr[3])
+        qc.cx(qr[2], qr[1])
+        qc.x(qr[1])
+        qc.cx(qr[2], qr[1])
+
+        qr = QuantumRegister(num_qubits + 1)
+        expected = QuantumCircuit(qr)
+
+        expected.cx(qr[2], qr[1])
+        expected.cx(qr[3], qr[4])
+        expected.cx(qr[1], qr[0])
+        expected.cx(qr[2], qr[3])
+        expected.cx(qr[2], qr[1])
+        expected.cx(qr[2], qr[1])
+
+        new_circuit = pm.run(qc)
+
+        self.assertEqual(Operator(expected), Operator(new_circuit))
```

Comment:
If you're already building the expected circuit, you could check the equality more efficient by just comparing the circuits
```suggestion
        self.assertEqual(expected, new_circuit)
```
You might have to remove the use of an explicit `QuantumRegister` and just construct the circuits as `QuantumCircuit(num_qubits + 1)`. (Also side remark, it's a bit weird to have `num_qubits` not actually be the number of qubits in the circuit, but you can keep it if you prefer this 🙂 )